clients/linux/src/filetree.rs

```diff
@@ -179,52 +179,36 @@ impl FileTree {
         m: &Messenger,
         c: &Arc<LbCore>,
     ) -> impl Fn(&TreeView, &DragContext, i32, i32, &SelectionData, u32, u32) {
-        closure!(m, c => move
-            |w, d, x, y, s, info, time| {
+        closure!(m, c => move |w, d, x, y, s, info, time| {
             if let Some((Some(mut path), pos)) = w.get_dest_row_at_pos(x, y) {
                 let model = w.get_model().unwrap().downcast::<TreeStore>().unwrap();
 
                 let mut parent = model.get_iter(&path).unwrap();
-                match pos {
-                    TreeViewDropPosition::Before |
-                    TreeViewDropPosition::After => {
-                        path.up();
-                        parent = model.get_iter(&path).unwrap();
-                    }
-                    _ => {
-                        if tree_iter_value!(model, &parent, 3, String) == format!("{:?}", FileType::Document) {
-                            path.up();
-                            parent = model.get_iter(&path).unwrap();
-                        }
-                    }
-                }
-
-                if tree_iter_value!(model, &parent, 3, String) == format!("{:?}", FileType::Document) {
+                if (pos == TreeViewDropPosition::Before || pos == TreeViewDropPosition::After) || tree_iter_value!(model, &parent, 3, String) == format!("{:?}", FileType::Document) {
```

Comment:
@SmailBarkouch did you test this and everything worked the same? You got rid of the duplicate code after the match statement, but parent is modified in the match statement after `path.up()`, and then the duplicate code is run again. Now that's not the case, so I just wanted to make sure that nothing broke.

Reply:
Yeah I get what you mean. The duplicate code was harmless since the `path.up()` statements in the match always made parent into a folder. So when it got to the if statement (the duplicate code) outside of the match it never got to run since the parent was already a folder. I tested it again just to be sure and there are no problems.